### PR TITLE
[SYSTEMML-1991] Implementation of Sobol quasi random sequence generator

### PR DESCRIPTION
**Pull request by j143**
**Sobol Quasi Random Sequence generator**
This outputs a `matrix`, of `points` of shape `NxD`. where `N` - no.of points, `D` - no. of dimensions.

_How this works?_
1. This script contains precomputed direction numbers for uniform sampling.
2. Using these direction numbers further the direction vectors will be scaled.
3. Further with help of a `Bitwise_xor` operator, it computes the points & scales them down.

### CODE REVIEW

File: scripts/staging/sobol.dml [hunk 1]
```diff
@@ -0,0 +1,1084 @@
...
+      941     13      3013    1 3 1 7 29 3 47 237 157 769 839 3199 3195
+      942     13      3018    1 1 3 15 25 39 63 15 111 857 881 1505 7671
+      943     13      3020    1 1 7 1 3 35 41 215 99 895 1025 1483 4707
+      944     13      3025    1 3 5 1 1 31 25 247 113 841 397 1825 6969
+      945     13      3042    1 1 3 5 19 41 49 243 225 973 241 175 1041
+      946     13      3047    1 1 1 7 15 15 105 141 83 75 1675 3523 5219
+      947     13      3048    1 1 7 5 13 27 47 199 445 841 959 1157 2209
+      948     13      3051    1 3 5 15 23 31 31 81 85 33 785 2639 7799
+      949     13      3054    1 1 5 13 21 3 47 99 235 943 1731 2467 7891
+      950
```
Comment (j143) [inline scripts/staging/sobol.dml]: Hi @mboehm7 , this line uses `bitwise_xor` operation, can I implement a `bitwise_xor`. If yes, any preference as to which components should I change.s

**Comment by mboehm7** [comment]
@j143 if you like I can take care of SYSTEMML-1931 and you can use that as a blueprint for implementing the bitwise operations. What do you think?

**Comment by j143** [comment]
Hi @mboehm7 , thanks for the word. I am handling both the tasks, & till I'm halfway through.

**Comment by akchinSTC** [comment]
Build failed, see build log for details

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2225/<h2>Failed Tests: <span class='status-failure'>8</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2225/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>8</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2225/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.reorg/FullReverseTest/testReverseVectorSparseMR/'><strong>org.apache.sysml.test.integration.functions.reorg.FullReverseTest.testReverseVectorSparseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2225/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.reorg/FullReverseTest/testReverseVectorSparseSP/'><strong>org.apache.sysml.test.integration.functions.reorg.FullReverseTest.testReverseVectorSparseSP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2225/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.reorg/FullReverseTest/testReverseVectorDenseMR/'><strong>org.apache.sysml.test.integration.functions.reorg.FullReverseTest.testReverseVectorDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2225/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.reorg/FullReverseTest/testReverseVectorDenseSP/'><strong>org.apache.sysml.test.integration.functions.reorg.FullReverseTest.testReverseVectorDenseSP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2225/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.reorg/FullReverseTest/testReverseMatrixSparseMR/'><strong>org.apache.sysml.test.integration.functions.reorg.FullReverseTest.testReverseMatrixSparseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2225/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.reorg/FullReverseTest/testReverseMatrixSparseSP/'><strong>org.apache.sysml.test.integration.functions.reorg.FullReverseTest.testReverseMatrixSparseSP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2225/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.reorg/FullReverseTest/testReverseMatrixDenseMR/'><strong>org.apache.sysml.test.integration.functions.reorg.FullReverseTest.testReverseMatrixDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2225/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.reorg/FullReverseTest/testReverseMatrixDenseSP/'><strong>org.apache.sysml.test.integration.functions.reorg.FullReverseTest.testReverseMatrixDenseSP</strong></a></li></ul>

**Comment by j143** [comment]
preliminary script.

**Pull request closed without merging**